app/assets/javascripts/discourse/app/components/async-category-link.gjs

```diff
@@ -0,0 +1,33 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import didUpdate from "@ember/render-modifiers/modifiers/did-update";
+import categoryLink from "discourse/helpers/category-link";
+import Category from "discourse/models/category";
+
+export default class AsyncCategoryLink extends Component {
```

Comment:
Overall this makes sense. But naming is tricky, I think in the long run we don't necessarily want to keep `Async` in the name. Can we use `CategoryLink`? There might be a naming conflict with the helper, not sure. 

Would be great to also test in how many places we can replace the helper with this component. 

One more thing: it would be good to have this represented in the styleguide. The category badge screen there needs some love anyway.